build/dependencies.yaml

```diff
@@ -18,7 +18,7 @@ dependencies:
 
   # agnhost: bump this one first
   - name: "agnhost"
-    version: "2.29"
+    version: "2.30"
```

Comment:
I forget how this works but technically we can't use 2.30 until it's been promoted. Usually there's a followup PR that updates test/images/manifest once promotion has occurred. Is that the more appropriate PR to make this change?

The two-step image promotion dance and fixed/hardcoded version approach used by e2e test images could really stand to be revisited. Not suggesting we try to do it in v1.21 just more thinking out loud. Auto-tagging with date+gitsha is an approach that's worked well for other images in this project AFAIK.

Reply:
That makes sense. Let me modify the PR.

Reply:
Also opened https://github.com/kubernetes/kubernetes/issues/100273

Reply:
Using https://github.com/kubernetes/kubernetes/pull/98639/files as a reference, it looks like I will need to bump agnhost's version in 3 places with the same PR:
- build/dependencies.yaml
- test/images/agnhost/VERSION
- test/images/agnhost/agnhost.go